### ISSUE TITLE
Failed running fix command

### ISSUE THREAD

**Issue by lloricode** [nunomaduro/phpinsights#564]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| Library version  | 2.1.0

Hi!,

after running `composer u`

```
vagrant@homestead:~/code/laravel/project-x$ composer u
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Package operations: 0 installs, 3 updates, 0 removals
  - Upgrading andreaselia/laravel-api-to-postman (v1.10.0 => v1.11.0): Extracting archive
  - Upgrading php-parallel-lint/php-parallel-lint (v1.3.1 => v1.3.2): Extracting archive
  - Upgrading composer/composer (dev-main 7328f5e => dev-main c9baeda): Extracting archive
Generating optimized autoload files
```

it return this error
```
> @php artisan insight --fix --ansi

 586/586 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ✨ Analysis Completed !

[2022-02-22 00:01:26] `/home/vagrant/code/laravel/project-x`


   TypeError 

  Cannot assign Composer\PartialComposer to property NunoMaduro\PhpInsights\Domain\ComposerLoader::$composer of type ?Composer\Composer

  at vendor/nunomaduro/phpinsights/src/Domain/ComposerLoader.php:30
     26▕             self::$currentCollector = $collector;
     27▕ 
     28▕             $io = new NullIO();
     29▕             $composerPath = ComposerFinder::getPath($collector);
  ➜  30▕             self::$composer = (new Factory())->createComposer($io, ComposerFinder::getPath($collector), false, null, false);
     31▕             self::$composer->setLocker(self::getLocker($composerPath, $io));
     32▕         }
     33▕ 
     34▕         return self::$composer;

      +22 vendor frames 
  23  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()
Script @php artisan insight --fix --ansi handling the insight-fix event returned with error code 1

```

this is working as expected until running `composer u`

thank you

**Comment by lloricode** [comment]
by running this command `composer require composer/composer:"dev-main#7328f5e"` it will be back to normal,

**Comment by cmgmyr** [comment]
fixed in #566
